spec/addressable/uri_spec.rb

```diff
@@ -5515,6 +5515,47 @@ def to_s
   end
 end
 
+describe Addressable::URI, "When set the basename " do
```

Comment:
I don't think that the description you have here really conveys the scenario you're testing. It's a good test case, but your description of the test context is too vague. It's also a test context that exists elsewhere in the test suite. In general you should append to an existing test context when it already exists. In this case, there are definitely already test contexts for e.g. `http://example.com` which you can add a new `it` block to.